relay-event-normalization/src/normalize/mod.rs

```diff
@@ -282,6 +288,71 @@ impl ModelCosts {
     }
 }
 
+/// Regex that matches version or date patterns at the end of a model name
+///
+/// Pattern breakdown (matches either):
+/// 1. Version pattern: [-_]v\d+[:\.]?\d*([:\-].*)?
+///    - Separator: [-_]
+///    - Version: v\d+[:\.]?\d* (must start with 'v': v1, v1.0, v1:0)
+///    - Optional trailing content: [:\-].*
+/// 2. Date pattern: ([-_@])?(\d{4}[-/\.]\d{2}[-/\.]\d{2}|\d{8})
+///    - Optional separator: [-_@]?
+///    - YYYY-MM-DD with -, /, or . OR YYYYMMDD
+///
+/// Examples matched:
+/// - "-v1:0" in "claude-sonnet-4-20250514-v1:0"
+/// - "_v2" in "model_v2"
+/// - "-v1.0" in "gpt-4-v1.0"
+/// - "-20250522" in "claude-4-sonnet-20250522"
+/// - "-2025-06-10" in "o3-pro-2025-06-10"
+/// - "@20241022" in "claude-3-5-haiku@20241022"
+static VERSION_OR_DATE_REGEX: LazyLock<Regex> = LazyLock::new(|| {
+    Regex::new(r"([-_]v\d+[:\.]?\d*([:\-].*)?|([-_@])?(\d{4}[-/\.]\d{2}[-/\.]\d{2}|\d{8}))$")
```

Comment:
Trying this out on regex101.com I get weird results.

Also I think `[\.]` doesn't do what you think it does.